includes/api/class-wc-rest-product-reviews-controller.php

```diff
@@ -14,14 +14,1143 @@
+
+		/*
+		 * This array defines mappings between public API query parameters whose
+		 * values are accepted as-passed, and their internal WP_Query parameter
+		 * name equivalents (some are the same). Only values which are also
+		 * present in $registered will be set.
+		 */
+		$parameter_mappings = array(
+			'reviewer'         => 'author__in',
+			'reviewer_email'   => 'author_email',
+			'reviewer_exclude' => 'author__not_in',
+			'exclude'          => 'comment__not_in',
+			'include'          => 'comment__in',
+			'offset'           => 'offset',
+			'order'            => 'order',
+			'per_page'         => 'number',
+			'product'          => 'post__in',
+			'search'           => 'search',
+			'status'           => 'status',
+		);
+
+		$prepared_args = array();
+
+		/*
+		 * For each known parameter which is both registered and present in the request,
+		 * set the parameter's value on the query $prepared_args.
+		 */
+		foreach ( $parameter_mappings as $api_param => $wp_param ) {
+			if ( isset( $registered[ $api_param ], $request[ $api_param ] ) ) {
+				$prepared_args[ $wp_param ] = $request[ $api_param ];
+			}
+		}
+
+		// Ensure certain parameter values default to empty strings.
+		foreach ( array( 'reviewer_email', 'search' ) as $param ) {
+			if ( ! isset( $prepared_args[ $param ] ) ) {
+				$prepared_args[ $param ] = '';
+			}
+		}
+
+		if ( isset( $registered['orderby'] ) ) {
+			$prepared_args['orderby'] = $this->normalize_query_param( $request['orderby'] );
+		}
+
+		if ( isset( $prepared_args['status'] ) ) {
+			$prepared_args['status'] = 'approved' === $prepared_args['status'] ? 'approve' : $prepared_args['status'];
+		}
+
+		$prepared_args['no_found_rows'] = false;
+		$prepared_args['date_query']    = array();
+
+		// Set before into date query. Date query must be specified as an array of an array.
+		if ( isset( $registered['before'], $request['before'] ) ) {
+			$prepared_args['date_query'][0]['before'] = $request['before'];
+		}
+
+		// Set after into date query. Date query must be specified as an array of an array.
+		if ( isset( $registered['after'], $request['after'] ) ) {
+			$prepared_args['date_query'][0]['after'] = $request['after'];
+		}
+
+		if ( isset( $registered['page'] ) && empty( $request['offset'] ) ) {
+			$prepared_args['offset'] = $prepared_args['number'] * ( absint( $request['page'] ) - 1 );
+		}
+
+		/**
+		 * Filters arguments, before passing to WP_Comment_Query, when querying reviews via the REST API.
+		 *
+		 * @since 3.5.0
+		 * @link https://developer.wordpress.org/reference/classes/wp_comment_query/
+		 * @param array           $prepared_args Array of arguments for WP_Comment_Query.
+		 * @param WP_REST_Request $request       The current request.
+		 */
+		$prepared_args = apply_filters( 'woocommerce_rest_product_review_query', $prepared_args, $request );
+
+		$query        = new WP_Comment_Query();
+		$query_result = $query->query( $prepared_args );
```

Comment:
This needs to only get product review comments. Currently it gets every kind of comments.

Reply:
Good catch, we used in the past `get_approved_comments()` and never tried to check for it, but for sure good to avoid pingback or anything else that maybe be included.